File: 8Strings/find_dupli_using_bitwise.cpp
#include <stdio.h>
#include <conio.h>
#include <iostream>

using namespace std;

int main(int argc, char const *argv[])
{
    char x[]="finding";
    long H = 0,a = 0;
    for (int i = 0; x[i] != '\0'; i++)
    {
        a=1;
        a=a<<(x[i]-97);
        if((H&a)>0){
            cout<<x[i]<<" Duplicated"<<endl;
        }
        else
        {
            H=H|a;
        }
    }
    
    return 0;
}

File: 8Strings/find_dupli_using_bitwise.java
public class find_dupli_using_bitwise {
    public static void main(String[] args) {
        String x = "finding";
        long H = 0,a = 0;
        for (int i = 0; i < x.length(); i++)
        {
            a=1;
            a=a<<(x.charAt(i)-97);
            if((H&a)>0){
                System.out.println(x.charAt(i)+" duplicated");
            }
            else
            {
                H=H|a;
            }
        }
    }
    
}
